src/halmos/sevm.py

```diff
@@ -815,7 +864,7 @@ def sha3_data(self, data: Bytes, size: int) -> Word:
         sha3_expr = f_sha3(data)
 
         # assume hash values are sufficiently smaller than the uint max
-        self.solver.add(ULE(sha3_expr, con(2**256 - 2**64)))
+        self.path.append(ULE(sha3_expr, con(2**256 - 2**64)))
```

Comment:
do you think it would be useful to expose an additional method like `add_assertion`? It feels like this is a way to add it to the solver assertions but as a side effect it is also appended to the list of human-readable path conditions, when it doesn't really correspond to a path (branching) condition

Reply:
or just `path.assert(cond)` for short

Reply:
good point!  added a todo note to get back to this later.